server/api.go

```diff
@@ -382,25 +382,33 @@ func (d dexAPI) RevokeRefresh(ctx context.Context, req *api.RevokeRefreshReq) (*
 	return &api.RevokeRefreshResp{}, nil
 }
 
-func (d dexAPI) GetVehiclePrivilegeToken(ctx context.Context, req *api.GetVehiclePrivilegeTokenReq) (*api.GetVehiclePrivilegeTokenResp, error) {
-	expiry := d.serverConfig.Now().Add(time.Minute * 10) // TODO - Discuss an appropriate time
-	v := vehicleIDTokenClaims{
-		Issuer:     d.serverConfig.Issuer,
-		Subject:    req.VehicleTokenId,
-		Expiry:     expiry.Unix(),
-		IssuedAt:   d.serverConfig.Now().Unix(),
-		Privileges: req.PrivilegeIds,
-		UserID:     req.UserId,
-	}
-	payload, err := json.Marshal(v)
+func (d dexAPI) GetCustomToken(ctx context.Context, req *api.SignTokenRequest) (*api.SignTokenResp, error) {
+	issuedAt := d.serverConfig.Now()
+	expiry := issuedAt.Add(d.serverConfig.IDTokensValidFor)
+
+	baseClaims := map[string]any{
+		"sub": req.Subject,
+		"iat": issuedAt.Unix(),
+		"exp": expiry.Unix(),
+	}
+
+	customClaims := req.CustomClaims.AsMap()
```

Comment:
The naming here is tricky because this starts as the set of custom claims and becomes the entire set of claims.